plugins/woocommerce/includes/class-wc-mobile-messaging-handler.php

```diff
@@ -0,0 +1,207 @@
+	 */
+	private static function is_order_in_person_payment_eligible( WC_Order $order ): bool {
+		return in_array( $order->get_status(), array( 'pending', 'on-hold', 'processing' ), true ) &&
+			   in_array( $order->get_payment_method(), array( 'cod', 'woocommerce_payments', 'none' ), true ) && // phpcs:ignore WordPress.WhiteSpace.PrecisionAlignment.Found
+			   self::is_store_in_person_payment_eligible() && // phpcs:ignore WordPress.WhiteSpace.PrecisionAlignment.Found
+			   ! $order->is_paid() && // phpcs:ignore WordPress.WhiteSpace.PrecisionAlignment.Found
+			! empty(
+				$order->get_refunds()
+			);
+	}
+
+	/**
+	 * Returns the closest date of last usage of any mobile app platform.
+	 *
+	 * @return DateTime|null
+	 */
+	private static function get_closer_mobile_usage_date(): ?DateTime {
+		$mobile_usage = WC_Tracker::get_woocommerce_mobile_usage();
+
+		$last_ios_used     = self::get_last_used_or_null( 'ios', $mobile_usage );
+		$last_android_used = self::get_last_used_or_null( 'android', $mobile_usage );
+
+		return max( $last_android_used, $last_ios_used );
+	}
+
+	/**
+	 * Returns last used date of specified mobile app platform.
+	 *
+	 * @param string $platform     mobile platform to check.
+	 * @param array  $mobile_usage mobile apps usage data.
+	 *
+	 * @return DateTime|null last used date of specified mobile app
+	 */
+	private static function get_last_used_or_null(
+		string $platform, array $mobile_usage
+	): ?DateTime {
+		try {
+			if ( array_key_exists( $platform, $mobile_usage ) ) {
+				return new DateTime( $mobile_usage[ $platform ]['last_used'] );
+			} else {
+				return null;
+			}
+		} catch ( Exception $e ) {
+			return null;
+		}
+	}
+
+	/**
+	 * Checks if the store has specified country location and currency used.
+	 *
+	 * @param string $country country to compare store's country with.
+	 * @param string $currency currency to compare store's currency with.
+	 *
+	 * @return true if specified country and currency match the store's ones. false otherwise
+	 */
+	public static function has_store_specified_country_currency( string $country, string $currency ): bool {
+		return ( WC()->countries->get_base_country() === $country && get_woocommerce_currency() === $currency );
+	}
+
+
+	/**
+	 * Prepares message with a deep link to mobile payment.
+	 *
+	 * @param int $blog_id blog id to deep link to.
+	 * @param int $order_id order id to deep link to.
+	 *
+	 * @return string formatted message
+	 */
+	public static function accept_payment_message( int $blog_id, int $order_id ): string {
+		$deep_link_url = add_query_arg(
+			array(
+				'blog_id'  => absint( $blog_id ),
+				'order_id' => absint( $order_id ),
+			),
+			'https://woocommerce.com/mobile/payments'
+		);
+
+		return sprintf(
+			wp_kses_data(
+			/* translators: %s: Email link */
+				__(
+					'<a href="%1$s">Accept payments</a> with a card reader in our mobile app\.</br><a href="%2$s">Learn more about In-Person Payments\.</a>',
+					'woocommerce'
+				)
```

Comment:
1. I would probably move the `translators` comment down to a line above the beginning of the string, to make sure it gets captured by the translation string parser.
    ```
	__(
		/* translators: %s: Email link */
		'<a href="%1$s">Accept payments</a>...'
    ```
2. I don't think the period characters need to be escaped in this string.
3. `</br>` should be `<br />` instead.

Reply:
> I would probably move the translators comment down to a line above the beginning of the string, to make sure it gets captured by the translation string parser.

I've tried doing this:

<img width="1382" alt="image" src="https://user-images.githubusercontent.com/5845095/187388453-874996c4-85ca-4706-aeb8-74d856715825.png">

and phpcs returns:

```
| ERROR | A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders. (WordPress.WP.I18n.MissingTranslatorsComment)
```

so I'm not sure we can do this.

Reply:
Oh interesting. I assumed the translators comment needed to be adjacent to the actual text string, but I guess it needs to be adjacent to the gettext function ( `__()` ) instead. So, you're already doing it right.